File: rmn-be/Core/Models/DiningTable.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SEP_Restaurant_management.Core.Models;

public partial class DiningTable
{
    [Key]
    public int TableId { get; set; }

    [Required]
    [MaxLength(30)]
    public string TableCode { get; set; } = default!;

    [MaxLength(100)]
    public string? TableName { get; set; }

    public int Capacity { get; set; }

    [Required]
    [MaxLength(20)]
    public string Status { get; set; } = "AVAILABLE";

    public bool IsActive { get; set; } = true;

    public virtual ICollection<ReservationTable> ReservationTables { get; set; } = new List<ReservationTable>();
    public virtual ICollection<Order> Orders { get; set; } = new List<Order>();
}

File: rmn-fe/src/types/models/dining-table.ts
export interface DiningTableResponse {
    tableId: number;
    tableCode: string;
    tableName?: string;
    capacity: number;
    status: "AVAILABLE" | "OCCUPIED" | "RESERVED" | "MAINTENANCE";
    isActive: boolean;
}

export interface CreateDiningTableRequest {
    tableCode: string;
    tableName?: string;
    capacity: number;
    status?: string;
}

export interface UpdateDiningTableRequest {
    tableCode?: string;
    tableName?: string;
    capacity?: number;
    status?: string;
    isActive?: boolean;
}

export interface UpdateReservationStatusRequest {
    status: string;
    tableIds?: number[];
}